<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>954</width>
    <height>702</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
    <property name="sizeConstraint">
     <enum>QLayout::SetMaximumSize</enum>
    </property>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QDoubleSpinBox" name="input_a">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="locale">
             <locale language="English" country="UnitedStates"/>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="prefix">
             <string>a=</string>
            </property>
            <property name="minimum">
             <double>-10000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="input_b">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="locale">
             <locale language="English" country="UnitedStates"/>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="prefix">
             <string>b=</string>
            </property>
            <property name="minimum">
             <double>-10000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>2.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QDoubleSpinBox" name="input_A">
            <property name="minimumSize">
             <size>
              <width>157</width>
              <height>33</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="locale">
             <locale language="English" country="UnitedStates"/>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="prefix">
             <string>A=</string>
            </property>
            <property name="minimum">
             <double>-10000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>-720.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="input_B">
            <property name="minimumSize">
             <size>
              <width>157</width>
              <height>33</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="locale">
             <locale language="English" country="UnitedStates"/>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="prefix">
             <string>B=</string>
            </property>
            <property name="minimum">
             <double>-10000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>720.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QSpinBox" name="input_points">
        <property name="minimumSize">
         <size>
          <width>166</width>
          <height>33</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>points=</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100000</number>
        </property>
        <property name="value">
         <number>5000</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="doubleSpinBox">
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>scale=</string>
        </property>
        <property name="decimals">
         <number>7</number>
        </property>
        <property name="maximum">
         <double>1000000000.000000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Restore</string>
        </property>
        <property name="shortcut">
         <string>R</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QSpinBox" name="x_spin">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
          <property name="prefix">
           <string>x=</string>
          </property>
          <property name="minimum">
           <number>-1000000</number>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="y_spin">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
          <property name="prefix">
           <string>y=</string>
          </property>
          <property name="minimum">
           <number>-1000000</number>
          </property>
          <property name="maximum">
           <number>1000000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="angle_spin">
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="suffix">
         <string>°</string>
        </property>
        <property name="minimum">
         <double>-180.000000000000000</double>
        </property>
        <property name="maximum">
         <double>180.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <widget class="GLWidget" name="openGLWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>533</width>
        <height>533</height>
       </size>
      </property>
      <property name="sizeIncrement">
       <size>
        <width>1</width>
        <height>1</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>533</width>
        <height>533</height>
       </size>
      </property>
      <property name="cursor">
       <cursorShape>CrossCursor</cursorShape>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="PointView" name="x_box">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>x:</string>
        </property>
        <property name="decimals">
         <number>5</number>
        </property>
        <property name="minimum">
         <double>-1000000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="PointView" name="y_box">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="prefix">
         <string>y:</string>
        </property>
        <property name="decimals">
         <number>5</number>
        </property>
        <property name="minimum">
         <double>-1000000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>954</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action">
   <property name="text">
    <string>Жопа сьела трусы</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>GLWidget.hpp</header>
   <slots>
    <signal>scale_changed(double)</signal>
    <signal>scale_is_small(QString)</signal>
    <signal>scale_is_normal()</signal>
    <signal>x_changed(int)</signal>
    <signal>y_changed(int)</signal>
    <signal>angle_changed(double)</signal>
    <signal>point_pos(QPointF)</signal>
    <slot>set_x(int)</slot>
    <slot>set_y(int)</slot>
    <slot>set_angle(double)</slot>
    <slot>restore()</slot>
    <slot>set_scale(double)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>PointView</class>
   <extends>QDoubleSpinBox</extends>
   <header>pointview.hpp</header>
   <slots>
    <slot>new_point(QPointF)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>input_B</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>myslot(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_a</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>myslot(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>16</x>
     <y>295</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_b</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>myslot(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>36</x>
     <y>326</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_A</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>myslot(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>57</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input_points</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>myslot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>angle_changed(double)</signal>
   <receiver>angle_spin</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>125</x>
     <y>281</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>angle_spin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_angle(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>623</x>
     <y>521</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>x_changed(int)</signal>
   <receiver>x_spin</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>58</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>openGLWidget</receiver>
   <slot>restore()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>scale_is_normal()</signal>
   <receiver>statusbar</receiver>
   <slot>clearMessage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>570</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>669</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>scale_changed(double)</signal>
   <receiver>doubleSpinBox</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>413</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>177</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>openGLWidget</receiver>
   <slot>set_scale(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>scale_is_small(QString)</signal>
   <receiver>statusbar</receiver>
   <slot>showMessage(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>544</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>674</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>point_pos(QPointF)</signal>
   <receiver>y_box</receiver>
   <slot>new_point(QPointF)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>740</x>
     <y>556</y>
    </hint>
    <hint type="destinationlabel">
     <x>761</x>
     <y>653</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>point_pos(QPointF)</signal>
   <receiver>x_box</receiver>
   <slot>new_point(QPointF)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>611</x>
     <y>546</y>
    </hint>
    <hint type="destinationlabel">
     <x>574</x>
     <y>645</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>y_changed(int)</signal>
   <receiver>y_spin</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>557</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>332</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>myslot()</slot>
  <slot>resize(int)</slot>
  <slot>coord(bool)</slot>
  <slot>myslot(double)</slot>
 </slots>
</ui>
